server/app/views/admin/programs/ProgramBlockEditView.java

```diff
@@ -645,6 +663,77 @@ private Optional<FormTag> optionalToggle(
             .with(optionalButton));
   }
 
+  private Optional<FormTag> addressCorrectionEnabledToggle(
+      Request request,
+      InputTag csrfTag,
+      long programDefinitionId,
+      long blockDefinitionId,
+      QuestionDefinition questionDefinition,
+      boolean addressCorrectionEnabled) {
+    if (!questionDefinition.isAddress()) {
+      return Optional.empty();
+    }
+    if (questionDefinition instanceof StaticContentQuestionDefinition) {
+      return Optional.empty();
+    }
+    String toolTipText =
+        "Enabling address correction will check the resident's address to ensure it is accurate.";
+    if (!featureFlags.isEsriAddressCorrectionEnabled(request)) {
+      toolTipText +=
+          " To use this feature, you will need to have your IT manager configure the GIS service.";
+    }
+    ButtonTag addressCorrectionButton =
+        TagCreator.button()
+            .withClasses(
+                "flex",
+                "gap-2",
+                "items-center",
+                addressCorrectionEnabled ? "text-black" : "text-gray-400",
+                "font-medium",
+                "bg-transparent",
+                "rounded-full",
+                StyleUtils.hover("bg-gray-400", "text-gray-300"))
+            .withType("submit")
+            .with(p("Address correction").withClasses("hover-group:text-white"))
+            .with(
+                div()
+                    .withClasses("relative")
+                    .with(
+                        div()
+                            .withClasses(
+                                addressCorrectionEnabled ? "bg-blue-600" : "bg-gray-600",
+                                "w-14",
+                                "h-8",
+                                "rounded-full"))
+                    .with(
+                        div()
+                            .withClasses(
+                                "absolute",
+                                "bg-white",
+                                addressCorrectionEnabled ? "right-1" : "left-1",
+                                "top-1",
+                                "w-6",
+                                "h-6",
+                                "rounded-full")))
+            .with(div(ViewUtils.makeSvgToolTip(toolTipText, Icons.HELP)));
+    String toggleAddressCorrectionAction =
+        controllers.admin.routes.AdminProgramBlockQuestionsController.setAddressCorrectionEnabled(
+                programDefinitionId, blockDefinitionId, questionDefinition.getId())
+            .url();
+    return Optional.of(
+        form(csrfTag)
+            .withMethod(HttpVerbs.POST)
+            .withCondOnsubmit(
+                !featureFlags.isEsriAddressCorrectionEnabled(request), "return false;")
```

Comment:
question (non-blocking): rather than include this `return false;` bit could we instead not render this code at all if the feature isn't enabled?

Reply:
That's how I originally had it (before referencing the mocks) and last week during a checkin meeting I asked @mferhut if we want to keep it disabled like in the mocks or not show it and her preference was to keep the toggle visible and disabled.

Reply:
Ah ok, thanks